src/type-hierarchy.ts

```diff
@@ -0,0 +1,213 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as vscodelc from 'vscode-languageclient';
+
+export namespace TypeHierarchyDirection {
+export const Children = 0;
+export const Parents = 1;
+export const Both = 2;
+}
+
+type TypeHierarchyDirection = 0|1|2;
```

Comment:
`TypeHierarchyDirection` is used for namespace above, can we use a different name for type?

Reply:
My understanding was that this pairing of a namespace and a type with the same name is a pattern in the vscode API.

This is how `TypeHierarchyDirection` is defined in the [type hierarchy protocol proposal](https://github.com/microsoft/vscode-languageserver-node/pull/426/commits/876d7fe224e17d7c08258a6da21d11a2df9c1d0d#diff-0b22db02bb4adb043694d9d0a11d0aefR83), as the pattern is used for other enum-like things in the existing API (e.g. [`TextDocumentSyncKind`](https://github.com/microsoft/vscode-languageserver-node/blob/bf091f5bff33797888aa8b14b09cae918e88f589/protocol/src/common/protocol.ts#L1036)).